assets/js/amp-customize-controls.js

```diff
@@ -0,0 +1,194 @@
+	 *
+	 * @param {string} url URL.
+	 * @return {boolean} whether it is an AMP URL.
+	 */
+	function isAmpUrl( url ) {
+		var urlParser = document.createElement( 'a' );
+		urlParser.href = url;
+		if ( /(^|\?|&)amp=1(?=&|$)/.test( urlParser.search ) ) {
+			return true;
+		}
+		return ( /\/amp\/?$/ ).test( urlParser.pathname );
+	}
+
+	/**
+	 * Create an non-AMP version of a URL.
+	 *
+	 * @param {string} url URL.
+	 * @return {string} non-AMPified URL.
+	 */
+	function unampifyUrl( url ) {
+		var urlParser = document.createElement( 'a' );
+		urlParser.href = url;
+		urlParser.pathname = urlParser.pathname.replace( /\/amp\/?$/, '' );
+		urlParser.search = urlParser.search.replace( /(^|\?|&)amp=1/, '' );
+		return urlParser.href;
+	}
+
+	/**
+	 * Create an AMP version of a URL.
+	 *
+	 * @param {string} url URL.
+	 * @return {string} AMPified URL.
+	 */
+	function ampifyUrl( url ) {
+		var urlParser = document.createElement( 'a' );
+		urlParser.href = unampifyUrl( url );
+		if ( urlParser.search.length ) {
+			urlParser.search += '&';
+		}
+		urlParser.search += 'amp=1';
+		return urlParser.href;
+	}
+
+	/**
+	 * Hook up all AMP preview interactions once panel is ready.
+	 *
+	 * @param {wp.customize.Panel} panel The AMP panel.
+	 * @return {void}
+	 */
+	function panelReady( panel ) {
+		var ampToggle = false;
+
+		/**
+		 * Make current URL AMPified if toggle is on.
+		 *
+		 * @param {string} url URL.
+		 * @return {string} AMPified URL.
+		 */
+		function setCurrentAmpUrl( url ) {
+			if ( ! ampToggle && isAmpUrl( url ) ) {
+				return unampifyUrl( url );
+			} else if ( ampToggle && ! isAmpUrl( url ) ) {
+				return ampifyUrl( url );
+			}
+			return url;
+		}
+
+		/**
+		 * Swap to AMP version of URL in preview.
+		 *
+		 * @return {void}
+		 */
+		function toggleAmpView() {
+			api.previewer.previewUrl.set( setCurrentAmpUrl( api.previewer.previewUrl.get() ) );
+		}
+
+		/**
+		 * Inspect iFrame metadata to see if loaded page was successfully AMP'd.
+		 *
+		 * @return {void}
+		 */
+		function isPageAmpified() {
+			var ampEnabled = false,
+				headContent = $( api.previewer.preview.iframe ).contents().find( 'head' ).html();
+
+			if ( headContent.search( '<meta name="generator" content="AMP Plugin' ) >= 0 ) {
+				ampEnabled = true;
+			}
+
+			// Update tooltip display based on above IF it's currently toggled on.
+			if ( ampToggle && ! ampEnabled ) {
+				toggleControl( false, true, true );
+			} else if ( ! ampToggle && ! ampEnabled ) {
+
+				// Enable toggle but don't trigger anything else.
+				$( '.amp-toggle input' ).prop( 'disabled', false );
+			}
+		}
+
+		/**
+		 * Central function to control UI toggle elements.
+		 *
+		 * @param {bool} checked If toggle input should be checked.
+		 * @param {bool} disabled If toggle input should be disabled.
+		 * @param {bool} notification If tooltip notification should appear.
+		 * @return {void}
+		 */
+		function toggleControl( checked, disabled, notification ) {
```

Comment:
Can this function take an object as its sole param instead of three positional params? I think that will make it easier to read.

Reply:
This function can likely be removed altogether due to swapping to `wp.customize.state` 😄